webapp/src/app.tsx

```diff
@@ -1029,11 +1029,24 @@ export class ProjectView
             agreeIcon: "sign out",
             disagreeLbl: lf("Cancel")
         }).then(r => {
-            if (!r) return;
-            this.reload = true; //Indicate we are goint to reload next.
-            workspace.resetAsync()
-                .done(() => window.location.reload(),
-                () => window.location.reload())
+            if (!r) return Promise.resolve();
+            if (hf2Connection) {
+                return hf2Connection.disconnectAsync()
+                .then(() => {
+                    this.reload = true; //Indicate we are going to reload next.
```

Comment:
Move this code block into a function so that it doesn't need to be duplicated below

Reply:
+1